test/adaptive_load/scoring_function_test.cc

```diff
@@ -0,0 +1,180 @@
+#include "external/envoy/source/common/config/utility.h"
+
+#include "adaptive_load/scoring_function_impl.h"
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+
+namespace {
+
+TEST(BinaryScoringFunctionConfigFactoryTest, GeneratesEmptyConfigProto) {
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<ScoringFunctionConfigFactory>(
+          "nighthawk.binary");
+  const Envoy::ProtobufTypes::MessagePtr message = config_factory.createEmptyConfigProto();
+  const nighthawk::adaptive_load::BinaryScoringFunctionConfig expected_config;
+  EXPECT_EQ(message->DebugString(), expected_config.DebugString());
+  EXPECT_TRUE(Envoy::MessageUtil()(*message, expected_config));
+}
+
+TEST(LinearScoringFunctionConfigFactoryTest, GeneratesEmptyConfigProto) {
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<ScoringFunctionConfigFactory>(
+          "nighthawk.linear");
+  const Envoy::ProtobufTypes::MessagePtr message = config_factory.createEmptyConfigProto();
+  const nighthawk::adaptive_load::LinearScoringFunctionConfig expected_config;
+  EXPECT_EQ(message->DebugString(), expected_config.DebugString());
+  EXPECT_TRUE(Envoy::MessageUtil()(*message, expected_config));
+}
+
+TEST(BinaryScoringFunctionConfigFactoryTest, CreatesBinaryScoringFunctionFactory) {
```

Comment:
can we test the negative case were it throws.

Reply:
Done